## Understanding the data

Multiple sclerosis is the most common immune-mediated disorder affecting the central nervous system.

Nearly one million people have MS in the United States in 2022, and in 2020, about 2.8 million people were affected globally, with rates varying widely in different regions and among different populations.

The disease usually begins between the ages of 20 and 50 and is twice as common in women as in men.

MS was first described in 1868 by French neurologist Jean-Martin Charcot.

The name "multiple sclerosis" is short for multiple cerebro-spinal sclerosis, which refers to the numerous glial scars (or sclerae – essentially plaques or lesions) that develop on the white matter of the brain and spinal cord."

In [2]:
# libaries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline

In [4]:
# reading the data
df = pd.read_csv("multiple sclerosis.csv")

In [7]:
df.shape

(273, 20)

In [8]:
df.head()

Unnamed: 0  Gender  Age  Schooling  Breastfeeding  Varicella  \
0           0       1   34       20.0              1          1   
1           1       1   61       25.0              3          2   
2           2       1   22       20.0              3          1   
3           3       2   41       15.0              1          1   
4           4       2   34       20.0              2          1   

   Initial_Symptom  Mono_or_Polysymptomatic  Oligoclonal_Bands  LLSSEP  \
0              2.0                        1                  0       1   
1             10.0                        2                  1       1   
2              3.0                        1                  1       0   
3              7.0                        2                  1       0   
4              6.0                        2                  0       1   

   ULSSEP  VEP  BAEP  Periventricular_MRI  Cortical_MRI  Infratentorial_MRI  \
0       1    0     0                    0             1                   0   
1       0    1     0                    0             0                   0   
2       0    0     0                    0             1                   0   
3       1    1     0                    1             1                   0   
4       0    0     0                    1             0                   0   

   Spinal_Cord_MRI  Initial_EDSS  Final_EDSS  group  
0                1           1.0         1.0      1  
1                1           2.0         2.0      1  
2                0           1.0         1.0      1  
3                0           1.0         1.0      1  
4                0           1.0         1.0      1

## Data description

ID: Patient identifier<br>
Age: Age of the patient (in years)<br>
Schooling: time the patient spent in school (in years)<br>
Gender: 1=male, 2=female<br>
Breastfeeding: 1=yes, 2=no, 3=unknown<br>
Varicella: 1=positive, 2=negative, 3=unknown<br>
Initial_Symptoms: 
> 1=visual<br>
> 2=sensory<br>
> 3=motor<br>
> 4=other<br>
> 5= visual and sensory<br>
> 6=visual and motor<br>
> 7=visual and others<br>
> 8=sensory and motor<br>
> 9=sensory and other<br>
> 10=motor and other<br>
> 11=Visual, sensory and motor<br>
> 12=visual, sensory and other<br>
> 13=Visual, motor and other<br>
> 14=Sensory, motor and other<br>
> 15=visual,sensory,motor and other<br>

Mono _or_Polysymptomatic: 1=monosymptomatic, 2=polysymptomatic, 3=unknown<br>
Oligoclonal_Bands: 0=negative, 1=positive, 2=unknown<br>
LLSSEP: 0=negative, 1=positive<br>
ULSSEP:0=negative, 1=positive<br>
VEP:0=negative, 1=positive<br>
BAEP: 0=negative, 1=positive<br>
Periventricular_MRI:0=negative, 1=positive<br>
Cortical_MRI: 0=negative, 1=positive<br>
Infratentorial_MRI:0=negative, 1=positive<br>
Spinal_Cord_MRI: 0=negative, 1=positive<br>
initial_EDSS:?<br>
final_EDSS:?<br>
> Group: 1=CDMS, 2=non-CDMS


In [15]:
df.tail()

Unnamed: 0  Gender  Age  Schooling  Breastfeeding  Varicella  \
268         268       2   31        8.0              3          1   
269         269       1   21       15.0              3          3   
270         270       2   19       12.0              3          3   
271         271       2   32       15.0              3          3   
272         272       2   77        6.0              3          3   

     Initial_Symptom  Mono_or_Polysymptomatic  Oligoclonal_Bands  LLSSEP  \
268              9.0                        2                  0       0   
269              5.0                        2                  1       0   
270             13.0                        2                  0       1   
271             15.0                        2                  1       1   
272              2.0                        1                  0       0   

     ULSSEP  VEP  BAEP  Periventricular_MRI  Cortical_MRI  Infratentorial_MRI  \
268       0    0     0                    0             0                   0   
269       0    0     0                    0             0                   0   
270       1    1     0                    0             0                   0   
271       1    1     0                    1             1                   1   
272       1    0     0                    0             0                   0   

     Spinal_Cord_MRI  Initial_EDSS  Final_EDSS  group  
268                0           NaN         NaN      2  
269                1           NaN         NaN      2  
270                1           NaN         NaN      2  
271                0           NaN         NaN      2  
272                0           NaN         NaN      2

In [12]:
# display all rows
pd.set_option('display.max_rows', None)

In [13]:
df

Unnamed: 0  Gender  Age  Schooling  Breastfeeding  Varicella  \
0             0       1   34       20.0              1          1   
1             1       1   61       25.0              3          2   
2             2       1   22       20.0              3          1   
3             3       2   41       15.0              1          1   
4             4       2   34       20.0              2          1   
5             5       1   29       22.0              1          1   
6             6       2   53       20.0              1          1   
7             7       2   24       15.0              1          1   
8             8       1   36       15.0              1          1   
9             9       2   28       20.0              1          1   
10           10       2   60       12.0              3          2   
11           11       2   25       20.0              1          1   
12           12       1   34       12.0              1          1   
13           13       1   36        0.0              1          1   
14           14       2   29       15.0              1          1   
15           15       1   29       12.0              1          2   
16           16       2   29       20.0              1          1   
17           17       2   24       20.0              1          1   
18           18       2   51        9.0              1          2   
19           19       2   36       20.0              1          1   
20           20       2   32       15.0              3          1   
21           21       2   30       15.0              3          2   
22           22       1   50       12.0              1          2   
23           23       1   38       15.0              1          2   
24           24       1   22       20.0              3          2   
25           25       1   34       15.0              2          2   
26           26       1   34       12.0              2          1   
27           27       2   19       15.0              2          1   
28           28       2   19       15.0              2          2   
29           29       1   24       15.0              2          1   
30           30       2   25       15.0              2          2   
31           31       2   27       20.0              2          2   
32           32       1   36       15.0              1          2   
33           33       1   38       15.0              2          1   
34           34       1   23       20.0              2          2   
35           35       2   22       15.0              1          1   
36           36       2   30       15.0              3          2   
37           37       1   37       20.0              3          2   
38           38       2   38       12.0              2          2   
39           39       1   65       20.0              1          2   
40           40       2   39       15.0              3          2   
41           41       1   21       20.0              2          2   
42           42       1   49       20.0              3          2   
43           43       2   35       15.0              1          1   
44           44       2   42       12.0              1          1   
45           45       2   28       12.0              1          2   
46           46       2   41        9.0              1          2   
47           47       2   34       20.0              1          1   
48           48       1   22       15.0              1          1   
49           49       2   32       20.0              1          1   
50           50       2   24       20.0              1          1   
51           51       2   19       15.0              2          2   
52           52       1   33       12.0              1          2   
53           53       2   31       15.0              2          2   
54           54       1   51        9.0              1          1   
55           55       1   19       15.0              3          1   
56           56       2   28       15.0              1          1   
57 

## Data at first glance

> Dimension - 273 rows, 20 columns

> Unnamed: 0 column which is of no use for the analysis